### DISCUSSION TITLE
AES encryptAesCbc Error

### DISCUSSION THREAD

**Discussion by windy-itus** [korlibs/krypto#44] [Q&A] [answered]
encryptAesCbc func in AES occurs indexoutofboundsexception with "words" val  has lengh = 5. Is this an error?
![image](https://user-images.githubusercontent.com/50759931/134453676-a88c9c3c-2fa2-4b0f-a08e-81fa220712d9.png)

**Comment by soywiz** [comment]
@bibinbodongti can you provide an example of usage that reproduces the problem?
Ideally if you can provide an `assertEquals`, that would be easier to add a test, fix it and prevent regressions

**Reply by windy-itus** [reply]
For Ex: 
I have data=[0, 0, 4, 1, 0, 0, 0, 0, 0, 14, 0, 0, 0, 14, 0, 1, 4, 1, 1, 0] (Byte[20]) 
and key = [17, 9, 12, 81, -92, 44, 11, -104, -51, 43, 123, 116, -128, -40, 13, -56, 100, 88, 37, -116, 1, 12, 
48, 110, 96, -23, 76, -34, 127, -114, -81, -5]  (Byte[32])
I call command: AES.encryptEes128Cbc(data, key).
Then, I receive java.lang.arrayindexoutofboundsexception: length=5; index=5 encryptaes128cbc

**Comment by soywiz** [comment, marked as answer]
Yeah it seems to be expected. But the error reported was a bit misleading. I have handled it here:

* https://github.com/korlibs/korge-next/commit/54f3207e26ea792a7fccf2d4291a9114dafe87ee
* https://github.com/korlibs/korge-next/commit/bd4e8672e5c904eda20ed49fbfe47a6358f24fb8

If the number of bytes you are providing in the data are not multiple of 16, you have to specify a padding algorithm.

For example:

```kotlin
AES.encryptEes128Cbc(data, key, padding = Padding.PKCS7Padding)
```

**Reply by windy-itus** [reply]
Awesome. Thank you very much
